operator/pkg/securedcluster/values/translation/image_overrides.go

```diff
@@ -4,10 +4,13 @@ import "github.com/stackrox/rox/operator/pkg/images"
 
 var (
 	imageOverrides = images.Overrides{
-		images.Main:          "image.main.fullRef",
-		images.CollectorSlim: "image.collector.slim.fullRef",
-		images.CollectorFull: "image.collector.full.fullRef",
-		images.ScannerSlim:   "image.scanner.fullRef",
-		images.ScannerSlimDB: "image.scannerDb.fullRef",
+		images.Main:             "image.main.fullRef",
+		images.CollectorSlim:    "image.collector.slim.fullRef",
+		images.CollectorFull:    "image.collector.full.fullRef",
+		images.ScannerSlim:      "image.scanner.fullRef",
+		images.ScannerSlimDB:    "image.scannerDb.fullRef",
+		images.ScannerV4DB:      "image.scannerV4.db.fullRef",
+		images.ScannerV4Indexer: "scannerV4.indexer.image.fullRef",
+		images.ScannerV4Matcher: "scannerV4.matcher.image.fullRef",
```

Comment:
There is definitely a lot of opportunity for confusion here.

tl;dr: I believe the values above should be in-line with the rest of the images listed, that is `image.scannerV4.fullRef` and `image.scannerV4DB.fullRef`.

Reason: See https://github.com/stackrox/stackrox/blob/14ec4c7db56d9e1248ad0e4bfd10400b6934793f/image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl#L292.

Internally, in the (shared) scanner V4 init code we expect the image configuration in `scannerV4.<component>.image`.

But to not break the consistency within the secured-cluster chart we expect the configurations be reside in `image.scannerV4` and `image.scannerV4DB` -- this is simply how it is being done there for all the other images.

Then we do this here: https://github.com/stackrox/stackrox/blob/14ec4c7db56d9e1248ad0e4bfd10400b6934793f/image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl#L297
which copies the input we get from the user of the Helm chart (from `image.scannerV4`) and copy it to the path `scannerV4....`.
So that the image configuration code here https://github.com/stackrox/stackrox/blob/14ec4c7db56d9e1248ad0e4bfd10400b6934793f/image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl#L81 can work with it.

Since here, in this case, the operator can be regarded as the user of the Helm chart, I believe it should be what I wrote in the tl;dr above.

Reply:
You're right. Thanks for the context Moritz I've changed the image reference and double checked they are correctly passed through.